### PR TITLE
Improve RecyclableMemoryStream usage

### PR DESCRIPTION
**Pull request by kerams**
- Made use of `RecyclableMemoryStreamManager` in XML serialization too
- Added an upper bound (<3.0) to Microsoft.IO.RecyclableMemoryStream because v2 [does not contain breaking changes that are really relevant here](https://github.com/microsoft/Microsoft.IO.RecyclableMemoryStream/blob/master/CHANGES.md)
- Wired up `RecyclableMemoryStreamManager` through DI so that users can provide their own fine-tuned instance of the manager

### CODE REVIEW

**Comment by kerams** [comment]
Hm, I've just realized the `stream.ToArray()` in `Xml.ISerializer` is problematic. Should I revert that part?

**Review by dustinmoris — DISMISSED** [review]
Thanks for the PR. I agree with this approach!

**Review by dustinmoris — COMMENTED** [review]
I changed the target branch to be `develop` and had to resolve a couple merge conflicts. One of them was the package reference to `Microsoft.IO.RecyclableMemoryStream`. Is the upper bound of `1.3.0` still correct as it is now?

**Comment by dustinmoris** [comment]
I tried to push more changes to fix the build but it doesn't let me push to this fork. I think GitHub normally allows that unless it was disabled when creating the PR. Have a look at [my changes](https://github.com/giraffe-fsharp/Giraffe/commit/815369d39eb63eedf643f706f60058bff8b9b52c). I think we should include them as part of this PR.

**Comment by kerams** [comment]
Weird, I'll sort it out in the evening. In the meantime could you please go through the comments above? Particularly the one about removing the recyclable stream from XML serializer (because `ToArray()` can throw based on how the manager is set up) and then the use case of having the manager in DI but wanting Giraffe to instantiate its own.

**Comment by kerams** [comment]
I see you've raised the lower bound to 2.2 in the pre-release version. It's fine, though strictly speaking 1.3 would have stilled worked without any issues I think.

File: src/Giraffe/Middleware.fs
Comment (kerams) [reply src/Giraffe/Middleware.fs]: @dustinmoris, I've incorporated your changes and this is the last thing that needs to be resolved as far as I am concerned.

Comment (dustinmoris) [reply src/Giraffe/Middleware.fs]: I am happy with the changes as they are now. To me it makes sense that Giraffe uses the registered `RecyclableMemoryStreamManager` by default. If someone wanted to use a different manager for Giraffe then they can still overwrite the default behaviour by registering an `Xml.ISerializer` and `Json.ISerializer` in their service registration function so I think people have all the flexibility they want. Does this answer the question or have I misunderstood the issue here?

Comment (kerams) [reply src/Giraffe/Middleware.fs]: That answers it, thanks. In that case this is ready for to go.

**Review by dustinmoris — APPROVED** [review]
LGTM

**Comment by dustinmoris** [comment]
@slang25 @TheAngryByrd One of you want to provide a second review in case we missed something? I don't want to circumvent the implemented review policy with my admin rights if not necessary :)

**Comment by dustinmoris** [comment]
Going to merge this so it can be part of the 6.0.0-alpha-* releases

**Pull request merged**
